plugins/inventory/proxmox.py

```diff
@@ -610,6 +613,9 @@ def _populate(self):
         # gather vm's in pools
         self._populate_pool_groups(hosts)
 
+        # removes the general node-group from the inventory list
+        if self.exclude_nodes: self.inventory.remove_group(nodes_group)
```

Comment:
```suggestion
        if self.exclude_nodes:
            self.inventory.remove_group(nodes_group)
```

Reply:
Also instead of doing this: why not *avoid* adding the `nodes_group` in the first place? That would be a lot cleaner. Now it could happen that `nodes_group` coincides with another group that the user added manually.

Reply:
This would be possible as well, I opted for the shortest way. I don't actually know how and if this could collide with manually generated groups.

Another approach could be, to remove each node at the end of the for-loop in line 612 with an if-clause. This would lead to an existing, but empty nodes_group. 

Not adding the node to the inventory in the first was not immediately possible for me, since the host is expected to exist in the host-list or as a group `self.inventory.add_child()`. But if this approach is more preferred, I will do that. But this is less trivial to create a fix.

Reply:
> This would be possible as well, I opted for the shortest way. I don't actually know how and if this could collide with manually generated groups.

You can have more than one inventory source in Ansible. So assume that someone uses a manual `hosts` file, and after that this inventory plugin.

> Another working approach is. to remove each node at the end of the node-specific for-loop in line 612 with an if-clause.

It's better to not add things in the first place, instead of removing them later. Especially when you cannot be sure that nobody else added something of that same name before (like another inventory source called *before* this plugin).

Reply:
I would like to implement the option differently. My problem is, that the call to import composite vars from node expects the given node to be in the host list:  
 `variables = self.inventory.get_host(node['node']).get_vars()`    
I could not find a way to recreate the composite variables, without adding the node to the host inventory.

Reply:
I'm not sure why you need to do that. You simply need to avoid the following calls:
```
self.inventory.add_group(nodes_group)
```
and
```
self.inventory.add_child(nodes_group, node['node'])
```
Then `nodes_group` isn't created by this plugin, and nothing is added to that group by this plugin.

Reply:
This is correct. But this will still add and leave the individual nodes in the inventory. Calling the proxmox plugin will then result in the nodes being in the `ungrouped` system-group and could interfered with playbooks calling this group or `all`.   
  
Not having the nodes in the inventory is the primary concern of this PR.

Reply:
But that's not what your PR is doing. Your PR is still adding the nodes, it's only removing the group in the end. But removing the group does not remove its members from the inventory, these are kept. They are just no longer associated to the group that was removed.

Reply:
Yes, this PR is removing the nodes-group at the end of the `_populate` function. But as you see my example output, my code apparently not only removes the group, but it's members as well. Therefore, the existing nodes are removed from the inventory.

I just tested this through `ansible-inventory --list` and `ansible-inventory --graph` and can confirm this behavior. Removing the group `@proxmox_nodes` removes the group as well as the node from the inventory. After removing the group, the node is not in `@ungrouped`, it's deleted from the whole inventory.

Should I change my PR and remove each node at the end of it's for-loop, since this is a more straight-forward / less implicit way of removing the node-hosts from the inventory? Or should I close this PR, since both approaches (remove host after the loop or remove the whole group) are not sufficient to be merged in this plugin?

Reply:
If you check out the source of `InventoryData.remove_group` you can see that it does not remove the hosts that are part of this group: https://github.com/ansible/ansible/blob/devel/lib/ansible/inventory/data.py#L179-L188

In any case, you should rewrite your PR to never add hosts (and groups) that you don't want, instead of removing them later. That's the clean way to do this.